apisix/plugins/fault-injection.lua

```diff
@@ -64,6 +119,25 @@ local function sample_hit(percentage)
 end
 
 
+local function vars_match(vars, ctx)
+    local match_result
+    for _, var in ipairs(vars) do
+        local expr, err = expr.new(var)
+        if err then
+            core.log.error("vars expression does not match: ", err)
+            return nil, err
+        end
+
+        match_result = expr:eval(ctx.var)
+        if match_result then
+            break
+        end
+    end
+
+    return match_result, nil
+end
+
+
 function _M.check_schema(conf)
     local ok, err = core.schema.check(schema, conf)
```

Comment:
Better to check the expression here and add a test for it.

Reply:
updated.

Reply:
No, I mean checking the expression via:
```lua
        local expr, err = expr.new(var)
        if err then
            core.log.error("failed to create vars expression: ", err)
            return nil, err
        end
```

Reply:
added.